src/atom_table.rs

```diff
@@ -16,7 +16,7 @@ use modular_bitfield::prelude::*;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub struct Atom {
-    pub index: usize,
+    pub index: u64,
```

Comment:
I think these changes (`usize` to `u64`) would ideally be placed in their own commit, since they are orthogonal to the new possibility of making features optional, and so they can be reviewed separately.

Reply:
Agreed. Just was a bit lazy :)

Reply:
Accidentally Haskell instead of Prolog :-)

Reply:
What do you mean by "Accidentlly Haskell"? I didn't see any Haskell code... probably because I'm only a Haskell beginner (I'm a beginner of lots of things 8-))

Reply:
Haskell uses lazy evaluation ... so, it's a lazy language, while we are implementing Prolog here and have several different execution strategies at our disposal :-)

Reply:
Haha, I got your idea now! Yeah, different execution strategies is one of the main reason that attracted me to learn Prolog.